README.md

```diff
@@ -1,29 +1,131 @@
+
+To use this, we simply need to code `BoredPetsNFT is ERC721URISTorage`.
+
+- `@openzeppelin/contracts/utils/Counters.sol`
 
-## Front End
+In order to guarantee each `tokenId` is unique, we will increment it by one each time a NFT is minted. We use `Counters.Counter private _tokenIds` to increment and track the value of `tokenId`.
+
+Now, let's go over the methods:
+
+- `constructor() ERC721("Bored Pets Yacht Club", "BPYC")`
+
+The constructor is called when we deploy the smart contract. In this case, `constructor()` indicates our our NFT contract does not take in any parameters. Then, we set "Bored Pets Yacht Club" as the name and "BPYC" as the symbol for our NFTs.
+
+- `function mint(string memory _tokenURI) public`
+
+This funtion is super straightforward! Every time we mint an NFT, we want to guarantee `tokenId` is unique by incrementing the counter by one. Then, we just call `_safeMint` to mint and store the `_tokenURI`!
+
+## Write the deployment script
+
+In order to deploy the smart contract, we'll need to write our migrations file. Create the file `1_deploy_contracts.js` under `packages/truffle/migrations`. Then, copy and paste the code below:
+
+```javascript
+const BoredPetsNFT = artifacts.require("BoredPetsNFT");
+
+module.exports = function (deployer) {
+  deployer.deploy(BoredPetsNFT);
+};
+```
+`artifacts.require` is how we tell Truffle which contracts we'd like to interact with and returns a contract abstraction that we can use in the rest of our deployment script. NOTE: The name should match the contract name (`BoredPetsNFT`) and NOT the filename (`BoredPetsNFT.sol`).
+
+## Deploy the NFT smart contract
+
+This tutorial outlines how to deploy with Infura, but Truffle has great features to speed up local testing:
+
+1. Use Dashboards so you don't have to expose your private key
+2. Use Ganache forking to test using real data and interact with live contracts locally
+
+<TODO: Create guides for each of these>
+
+## Test the smart contract
 
 ...
 
-## Support
+## Community and Support
+
+Loved this tutorial and want to share with the Truffle community? Join our Discord and Reddit communities. Also, follow our Twitter [@trufflesuite](https://twitter.com/trufflesuite) or join our monthly community calls to stay up-to-date with the latest on all things Truffle.
 
 - [Discord](https://discord.com/invite/vbx6jy6XC8)
+- [Reddit](https://www.reddit.com/r/Truffle/)
+
+If you need help, find us on Github discussions!
+- [Truffle Support](https://github.com/orgs/trufflesuite/discussions)
+- [Ganache Support](https://github.com/orgs/trufflesuite/discussions/5121)
+
+Lastly, if you think you've discovered a bug, please let us know by creating an issue on the relevant repo or fixing it yourself - Truffle is open source :)
```

Comment:
Consider:

Lastly, if you you've discovered a bug, please let us know by creating an issue and/or creating a Pull Request with any fixes or contributions - All Truffle resources are open source :)